### PR TITLE
Issue #SC-1349 feat: TeacherId verification popup implementation

### PR DESCRIPTION
**Pull request by sudip-mukherjee**

### CODE REVIEW

**Comment by sudip-mukherjee** [comment]
> change navivateToResult -> navigateToResult in validate-teacher-identifier-popup.component.ts file.

Updated the changes.

File: src/app/client/src/app/app.component.html [hunk 1]
```diff
@@ -3,8 +3,8 @@
     [buttonLabel]="resourceService.frmelmnts.btn.submit" (submit)="updateFrameWork($event)"></app-popup>
   <app-tnc-popup (close)="onAcceptTnc()" #termsAndCondPopUp *ngIf="showTermsAndCondPopUp && !isOffline">
   </app-tnc-popup>
-  <app-validate-teacher-identifier-popup *ngIf="flag"> </app-validate-teacher-identifier-popup>
-  <app-user-location *ngIf="!isLocationConfirmed" [isCustodianOrgUser]="isCustodianOrgUser" [deviceProfile]="deviceProfile" [userProfile]="usersProfile">
+  <app-validate-teacher-identifier-popup *ngIf="showUserVerificationPopup" [userFeedData]="userFeed"> </app-validate-teacher-identifier-popup>
+  <app-user-location (close)="onLocationSubmit()" *ngIf="!isLocationConfirmed" [isCustodianOrgUser]="isCustodianOrgUser" [deviceProfile]="deviceProfile" [userProfile]="usersProfile">
```
Comment (rajeevsathish) [inline src/app/client/src/app/app.component.html]: why there is a code change in location popup?

File: src/app/client/src/app/app.component.html [hunk 2]
```diff
@@ -3,8 +3,8 @@
     [buttonLabel]="resourceService.frmelmnts.btn.submit" (submit)="updateFrameWork($event)"></app-popup>
   <app-tnc-popup (close)="onAcceptTnc()" #termsAndCondPopUp *ngIf="showTermsAndCondPopUp && !isOffline">
   </app-tnc-popup>
-  <app-validate-teacher-identifier-popup *ngIf="flag"> </app-validate-teacher-identifier-popup>
-  <app-user-location *ngIf="!isLocationConfirmed" [isCustodianOrgUser]="isCustodianOrgUser" [deviceProfile]="deviceProfile" [userProfile]="usersProfile">
+  <app-validate-teacher-identifier-popup *ngIf="showUserVerificationPopup" [userFeedData]="userFeed"> </app-validate-teacher-identifier-popup>
```
Comment (rajeevsathish) [inline src/app/client/src/app/app.component.html]: Do we need two variable to check can't we use the userFeed object for the if condition also?

File: src/app/client/src/app/app.component.ts [hunk 3]
```diff
@@ -470,4 +476,32 @@ setFingerPrintTelemetry() {
...
+  }
+
+  /** It will fetch user feed data if user is custodian as well as logged in. */
+  getUserFeedData() {
+    this.orgDetailsService.getCustodianOrg().subscribe(custodianOrg => {
+      if (this.userService.loggedIn &&
+        (_.get(this.userService, 'userProfile.rootOrg.rootOrgId') === _.get(custodianOrg, 'result.response.value'))) {
+          this.userService.getFeedData().subscribe(
+            (data) => {
+              const feedData = _.get(data, 'result.response.userFeed[0]');
```
Comment (rajeevsathish) [inline src/app/client/src/app/app.component.ts]: Why do we need a constant can't you assign it to the userFeed object?

File: src/app/client/src/app/app.component.ts [hunk 4]
```diff
@@ -470,4 +476,32 @@ setFingerPrintTelemetry() {
...
+              if (feedData && _.get(feedData, 'category').toLowerCase() === this.feedCategory.toLowerCase()) {
+                this.userFeed = feedData;
+                this.showUserVerificationPopup = true;
+              }
+              if (this.isLocationConfirmed) {
```
Comment (rajeevsathish) [inline src/app/client/src/app/app.component.ts]: why is this check valid?

File: src/app/client/src/app/app.component.ts [hunk 5]
```diff
@@ -470,4 +476,32 @@ setFingerPrintTelemetry() {
...
+                this.showUserVerificationPopup = true;
+              }
+            },
+            (error) => {
```
Comment (rajeevsathish) [inline src/app/client/src/app/app.component.ts]: please put some logs if any error happens

File: src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts [hunk 6]
```diff
@@ -1,47 +1,112 @@
...
+
   ngOnInit() {
-    this.instance = 'Tamil Nadu';
     this.userId = this.userService.userid;
+    this.processUserFeedData();
+    this.initializeFormField();
   }
 
+  initializeFormField() {
+    if (this.showStateDropdown) {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts]: this.userDetailsForm = new FormGroup({
        extId: new FormControl('', Validators.required)
      });
if( this.showStateDropdown ) {
userDetailsForm.addControl(name ,control);
}

File: src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts [hunk 7]
```diff
@@ -1,47 +1,112 @@
...
+    this.userDetailsForm.valueChanges.subscribe(val => {
       this.showError = false;
-      this.enableSubmitButton = true;
+      this.enableSubmitButton = (this.userDetailsForm.status === 'VALID');
+    });
+  }
+
+  verifyExtId(action: string) {
+    let request = {};
+    if (action === 'accept') {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts]: let req ={
request:{
'userId':this.userId,
'action':action,
'feedId':_.get(this.userFeedData, 'id')
}
}
if(condition){
req.request['userExtId']:-----,
req.request['channel']:-----
}

File: src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts [hunk 8]
```diff
@@ -1,47 +1,112 @@
...
+          'feedId': _.get(this.userFeedData, 'id')
+        }
+      };
     }
+    this.userService.userMigrate(request).subscribe((data) => {
+      if (_.get(data, 'responseCode') === 'OK' && _.get(data, 'result.response') === 'SUCCESS') {
+        this.extIdVerified = true;
+      }
+    }, (error) => {
+      if (_.get(error, 'responseCode') === 'invalidUserExternalId' && _.get(error, 'result.remainingAttempt') > 0) {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts]: use the .toLowercase()

File: src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts [hunk 9]
```diff
@@ -1,47 +1,112 @@
...
+        this.userDetailsForm.reset();
+        this.showError = true;
+      } else if (_.get(error, 'error.params.err') === 'USER_MIGRATION_FAILED') {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/shared-feature/components/validate-teacher-identifier-popup/validate-teacher-identifier-popup.component.ts]: use the .toLowercase()

**Pull request merged**